lnrpc/routerrpc/router_backend.go

```diff
@@ -699,6 +708,27 @@ func unmarshallHopHint(rpcHint *lnrpc.HopHint) (zpay32.HopHint, error) {
 	}, nil
 }
 
+// UnmarshalFeatures converts a list of uint32's into a valid feature vector.
+// This method checks that feature bit pairs aren't assigned toegether, and
+// validates transitive depdencies.
+func UnmarshalFeatures(rpcFeatures []lnrpc.FeatureBit) (*lnwire.FeatureVector, error) {
+	raw := lnwire.NewRawFeatureVector()
+	for _, bit := range rpcFeatures {
+		err := raw.SafeSet(lnwire.FeatureBit(bit))
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	fv := lnwire.NewFeatureVector(raw, lnwire.Features)
+	err := feature.ValidateDeps(fv)
```

Comment:
I'm not sure if invalid deps should stand in the way of invoice parsing.

Reply:
good point, we check this later on during pathfinding anyway. removed